toolchain/check/impl_lookup.cpp

```diff
@@ -1001,6 +1107,34 @@ auto EvalLookupSingleImplWitness(Context& context, SemIR::LocId loc_id,
       context, kind == FoundInFacet, query_self_const_id, *query_type_structure,
       query_specific_interface);
 
+  auto core_interface =
+      GetCoreInterface(context, query_specific_interface.interface_id);
+
+  // Wraps parameters for `LookupCppImpl`, with the difference being whether a
+  // best type structure is provided.
+  auto lookup_cpp_impl =
```

Comment:
Can we avoid the lambda? LookupDestroyCustomWitness has everything it needs to call LookupCppImpl directly.